### ISSUE TITLE
Rule Change: no-unused-vars should not flag function arguments used via the `arguments` object

### ISSUE THREAD

**Issue by EwanRoycroft** [eslint/eslint#16489]
### What rule do you want to change?

no-unused-vars

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
const test = function (myVar) {
    console.log(arguments);
}

test();
```

### What does the rule currently do for this code?

The variable `myVar` is not considered to have been used:
```shell
$ npx eslint index.js

index.js
  1:24  error  'myVar' is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

### What will the rule do after it's changed?

The variable `myVar` has been passed to another function using the `arguments` object, thus I propose that it should be considered to have been used.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Hi @EwanRoycroft, thanks for the issue!

I think the current behavior is correct. `myVar` is never referenced, so it can be removed and the function will still work the same.

```js
const test = function () {
    console.log(arguments);
}
```

**Comment by nzakas** [comment]
Agree with @mdjermanovic. `arguments` is array-like, and I don't think going into each item and marking it as used makes sense for this rule.
